app/models/history.rb

```diff
@@ -107,6 +107,169 @@ def self.record_automatically_closed(patient: nil, created_by: 'Sara Alert Syste
+
+    create_history(history[:patient], history[:created_by], HISTORY_TYPES[:monitoring_change], compose_message(history, field))
+  end
+
+  def self.case_status(history)
+    field = {
+      name: 'Case Status',
+      old_value: history[:patient][:case_status],
+      new_value: history[:params][:case_status]
+    }
+    return if field[:old_value] == field[:new_value]
+
+    if !history[:params][:monitoring].nil? && !history[:params][:monitoring]
+      history[:note] = ', and chose to "End Monitoring"'
+    elsif !history[:patient][:isolation].present? && history[:params][:isolation].present?
+      history[:note] = ', and chose to "Continue Monitoring in Isolation Workflow"'
+    end
+
+    create_history(history[:patient], history[:created_by], HISTORY_TYPES[:monitoring_change], compose_message(history, field))
+  end
+
+  def self.public_health_action(history)
+    field = {
+      name: 'Latest Public Health Action',
+      old_value: history[:patient][:public_health_action],
+      new_value: history[:params][:public_health_action]
+    }
+    return if field[:old_value] == field[:new_value]
+
+    create_history(history[:patient], history[:created_by], HISTORY_TYPES[:monitoring_change], compose_message(history, field))
+  end
+
+  def self.jurisdiction(history)
+    field = {
+      name: 'Jurisdiction',
+      old_value: Jurisdiction.find(history[:patient][:jurisdiction_id])[:path],
+      new_value: Jurisdiction.find(history[:params][:jurisdiction])[:path]
+    }
+    return if field[:old_value] == field[:new_value]
+
+    create_history(history[:patient], history[:created_by], HISTORY_TYPES[:monitoring_change], compose_message(history, field))
+  end
+
+  def self.assigned_user(history)
+    field = {
+      name: 'Assigned User',
+      old_value: history[:patient][:assigned_user],
+      new_value: history[:params][:assigned_user]
+    }
+    return if field[:old_value] == field[:new_value]
+
+    create_history(history[:patient], history[:created_by], HISTORY_TYPES[:monitoring_change], compose_message(history, field))
+  end
+
+  def self.pause_notifications(history)
+    field = {
+      name: 'notification status',
+      old_value: history[:patient][:pause_notifications] ? 'paused' : 'resumed',
+      new_value: history[:params][:pause_notifications] ? 'paused' : 'resumed'
+    }
+    return if field[:old_value] == field[:new_value]
+
+    create_history(history[:patient], history[:created_by], HISTORY_TYPES[:monitoring_change], compose_message(history, field))
+  end
+
+  def self.symptom_onset(history)
+    field = {
+      name: 'Symptom Onset Date',
+      type: 'date',
+      old_value: history[:patient][:symptom_onset]&.to_date&.strftime('%m/%d/%Y'),
+      new_value: history[:params][:symptom_onset]&.to_date&.strftime('%m/%d/%Y')
+    }
+    return if field[:old_value] == field[:new_value]
+
+    comment = compose_message(history, field)
+    comment += ' The system will now populate this date.' if new_value.nil?
```

Comment:
Found a bug due to this line when testing. `new_value` is undefined here. I think it should be `field[:new_value]`.

Reply:
thanks! fixed here: https://github.com/SaraAlert/SaraAlert/pull/371/commits/4dd09577c8ddca2afe844adf0f6d37e7a9d86fbf